todo-app/src/App.js

```diff
@@ -1,13 +1,56 @@
 import React, { Component } from 'react';
+import Task from './Task';
+import TaskForm from './TaskForm';
+import tasksStore from './TasksStore';
+import {observer} from 'mobx-react';
+
+@observer
+export default class App extends Component {
+
+  constructor(props) {
+		super(props);
+	
+    this.state = {
+	  	editingTaskID: null
+    };
 
-class App extends Component {
-  render() {
-    return (
-      <div>
-        Hello World!
-      </div>
-    );
   }
-}
 
-export default App;
+	handleTaskStartEdit = taskId => {
+		this.setState({editingTaskID: taskId});
+	};
+	
+	handleTaskStopEdit = () => {
+		this.setState({editingTaskID: null});
+	};
+	
+	
+  
+  render() {
+		return (
+	  <div>
+			<h1>Todo list</h1>
+			<h4 className='task-done'>✓ = {tasksStore.doneCount}</h4>
+			<h3>Enter description:</h3>
+			<TaskForm tasksStore = {tasksStore} />
+	  	{tasksStore.tasks.length === 0 ? 
+	  		<div>No items...</div> 
+	  	: 
+				<ul className='task-list'>
+					{
+						tasksStore.tasks.map(item => 
+						<Task 
+							key={item.id} 
+							id={item.id} 
+							tasksStore = {tasksStore}
```

Comment:
Because you are only using actions from the store inside your `Task` component, it might be nice to explicitly pass just those actions to `Task` so that it is clear what its requirements are.

MobX does do intelligent observation based on store values, but regardless you are still injecting a potentially large amount of useless data into each `Task` element on the page - imagine if we had a few hundred or thousand.